style: Style and structure the example project

nickcernis opened the pull request:
Adds styling, intro content, and examples of components, global and local CSS/Sass, and locally hosted fonts for the starter example.

For [BH-827](https://wpengine.atlassian.net/browse/BH-827).

<img width="1527" alt="Screenshot 2021-02-01 at 13 40 11" src="https://user-images.githubusercontent.com/647669/106460076-11690f00-6493-11eb-83b0-997b9d412e66.png">

### To test
Follow the [Lerna setup](https://github.com/wpengine/headless-framework/blob/canary/docs/DEVELOPMENT.md#npm-packages) and update your `.env.local` file.

You should see the screenshot above when you visit the homepage.

You should see the expected content on the homepage whether Settings → Reading displays latest posts or a static page, due to https://github.com/wpengine/headless-framework/pull/109/commits/5f26f23a93ad7a9fe347e66b3a964a0a56f0cd8b.

### Notes
Menu links are currently hard-coded in Header.tsx.

nickcernis committed:
chore: Add normalize.css and sass to starter theme

nickcernis committed:
chore: Add a main Sass file for global CSS

nickcernis committed:
chore: Load normalize.css and the main CSS file

nickcernis committed:
chore: Add fonts and font-face declarations

nickcernis committed:
design: Add base styling

nickcernis committed:
feat: Add simple stateless components and styles

nickcernis committed:
chore: Add image for homepage hero background

nickcernis committed:
feat: Add front page content and styles

nickcernis committed:
refactor: Use Next Link for site title

nickcernis committed:
refactor: Make Posts intro prop optional

nickcernis committed:
feat: Add Header and Footer to all page templates

nickcernis committed:
style: Add padding around page content

nickcernis committed:
style: Add breakpoint Sass variables

nickcernis committed:
docs: Mention the components scoped CSS styles

nickcernis committed:
feat: Give Hero component an ID prop

So that styling can be overridden on a per-page
basis.

The page using the Hero component can pass the
id prop and then override default Hero styling
in its [page-name].module.scss file.

nickcernis committed:
style: Give posts a scoped 'single' class

To help with responsive styling.

nickcernis committed:
style: Adjust styling at smaller widths

trevanhetzel commented on examples/preview/sass/_font-face.scss:
```diff
@@ -0,0 +1,143 @@
+}
+
+@font-face {
+	font-display: swap;
```
Couple things on the font:
1) Do we need _all_ the different weights? I typically only include the ones needed like regular, bold and italic.
2) In an effort to keep the amount of files this starter theme adds to a minimum, what do you think of just adding a `link` tag in the `head` to Google's hosted fonts? https://fonts.google.com/specimen/Public+Sans

nickcernis replied:
Good points, thanks.

I'm good with linking to Google Fonts instead. It looks like we'll need to amend `WPHead` to pass children to `Head` before I can do that, though (we don't have a good way to add extra elements to `<head>` at the moment). I'll spin up a quick PR to do that.

nickcernis replied:
Swapped local fonts for Google Fonts in https://github.com/wpengine/headless-framework/pull/109/commits/af85a5d0b5c439f49ff91be8de740aa4bb7d914e.

trevanhetzel commented on examples/preview/sass/components/CTA.module.scss:
```diff
@@ -0,0 +1,43 @@
+// CSS for the CTA component in components/CTA.tsx.
+
+@import "sass/variables";
```
Nit: since you're using `.scss`, the more common folder naming that I've seen for these files seems to be `/scss` rather than '/sass`

nickcernis replied:
Thanks! Done in https://github.com/wpengine/headless-framework/pull/109/commits/b46b61eb69c04b930402d8a1e9a68d2df23ce662.

mindctrl commented:
I get the following error when running the build (using `npm run bootstrap` per instructions): 

```
Failed to compile.

./components/CTA.tsx:23:10
Type error: Type '{ children: string; className: string; }' is not assignable to type 'IntrinsicAttributes'.
  Property 'children' does not exist on type 'IntrinsicAttributes'.

  21 |     <section className={styles.cta}>
  22 |       <div className={styles.wrap}>
> 23 |         <Heading className={styles.title}>{title}</Heading>
     |          ^
  24 |         <div className={styles.intro}>
  25 |           <div className={styles.children}>{children}</div>
  26 |           {buttonText && buttonURL && (
```

nickcernis committed:
chore: Pass heading levels in a type-safe way

Some hoop-jumping to satisfy TypeScript's type
checker. Fixes the type error that @mindctrl
kindly reported:

Type '{ children: string; className: string; }'
is not assignable to type 'IntrinsicAttributes'.

nickcernis commented:
> I get the following error…

Thanks for catching that, @mindctrl. Fixed in https://github.com/wpengine/headless-framework/pull/109/commits/3c4bdd0b3f539843dd7c351084d161fa30aec96a.

nickcernis committed:
refactor: Rename sass folder to scss

nickcernis committed:
chore: Fix import order

nickcernis committed:
refactor: Use Google Fonts instead of local fonts

Addresses feedback from @trevanhetzel at
https://github.com/wpengine/headless-framework/pull/109#discussion_r568136700.

nickcernis committed:
chore: Annotate function return types

nickcernis committed:
docs: Update font info on the front page.

To mention Google Fonts in place of locally hosted
fonts.

nickcernis committed:
fix: Incorrect nesting of Post heading and link

This was causing a console error for the Posts
component:

“Function components cannot be given refs.”

nickcernis committed:
chore: Suppress no-spreading warning

nickcernis committed:
chore: Suppress unneeded eslint rules

nickcernis committed:
fix: Load front-page.tsx if showing latest posts

If Settings → Reading is set to display latest
posts on the homepage (the default), this ensures
that the latest-posts template loads to display
the welcome content.

Before, users would have to set a static page
as the homepage in WP before they saw the default
homepage, which is a worse setup experience.

nickcernis committed:
chore: Remove unused file

nickcernis committed:
Merge branch 'canary' into theme-design

nickcernis committed:
chore: Inline Post type

The WPGraphQL types do not appear to be exposed,
so attempting to import { WPGraphQL } from
'@wpengine/headless' results in:

Type error: Module '"../node_modules/@wpengine/headless/dist"'
has no exported member 'WPGraphQL'.

mindctrl reviewed (approved):
Much nicer visuals! This LGTM. You've documented the trade offs and have them marked for a later visit. I'm ok if we merge this to unblock the other work, some related to the things you noted here.

claygriffiths committed:
fix: return '404' as possible template if there's a 404

claygriffiths committed:
feat: add 404.tsx template to examples/preview

claygriffiths committed:
fix: force $wp_query->parse_query() to run when getting templates

trevanhetzel commented on examples/preview/theme/front-page.tsx:
```diff
@@ -0,0 +1,134 @@
+          id={styles.home_hero}>
+          <p>
+            WP&nbsp;Engine’s Headless WordPress Framework includes this starter
+            theme, the{' '}
```
"includes this example project".

I want to avoid calling it a starter theme for now

nickcernis replied:
Makes sense! Changed in https://github.com/wpengine/headless-framework/pull/109/commits/63df63cb4c37f4e7113b88f695d91feb413bde82.

trevanhetzel commented on examples/preview/theme/front-page.tsx:
```diff
@@ -0,0 +1,134 @@
+        </Hero>
+        <section className={styles.explore}>
+          <div className="wrap">
+            <h2>Explore the Starter Theme</h2>
```
I think we should be calling this an example project, not a theme at this point

nickcernis replied:
Agreed. Fixed in Changed in https://github.com/wpengine/headless-framework/pull/109/commits/63df63cb4c37f4e7113b88f695d91feb413bde82.

trevanhetzel commented on examples/preview/theme/front-page.tsx:
```diff
@@ -0,0 +1,134 @@
+          count={6}
+          id={styles.post_list}
+        />
+        <CTA
```
I dig this

trevanhetzel reviewed (commented):
Reviewed the copy. I think it all works well! I just have a couple comments about calling it ann example project vs a starter theme

nickcernis committed:
fix: Remove front page template loader from index

Clay's fix in https://github.com/wpengine/headless-framework/pull/109/commits/9e996a96be95d042a48df69063d3e2aebb84aebf
ensures the front-page template loads for “latest
posts” homepages now, so front-page.tsx
will load automatically as expected. 🎉

nickcernis committed:
docs: change “theme” to “project” in welcome text

To address Trevan's copy edits:
- https://github.com/wpengine/headless-framework/pull/109#discussion_r568981682
- https://github.com/wpengine/headless-framework/pull/109#discussion_r568982025
